Automattic/wp-calypso
# Storage Upgrade: update offer

### Changes proposed in this Pull Request

This PR updates the products that are shown on the storage upgrade page to show Backup addons only (instead of plans).

Fixes 1182633034225700-as-1204031357952912.

### Implementation notes

It's kind of hard to comment on all changes since the code required a fair amount of changes to account for add-ons. Here are some pointer that I hope will help:
- Defined upgrade paths in the `UPGRADES`  map `use-get-storage-upgrade-products.ts`.
- Added new hooks in `packages/calypso-products/src/translations.tsx` to facilitate translations of storage amount.
- Removed the duration check from the storage page, since the default never changed anyway (the route doesn't consider duration). Upgrade relationships are defined in the `UPGRADES` map instead.
- Moved all copy in the `useGetProductCardData` hook.
- Removed the mentions of storage amount and backup days in the heading and subheading since they now depend on the add-ons being shown. This could be added back in a subsequent PR.

### Testing instructions

- Make sure you have a Jetpack site with a monthly Jetpack Backup 10 GB subscription
- Spin up Calypso by running this branch locally, or clicking on a live link below
- Visit `plans/storage/:site`
- Check that you see the 10GB, 100GB, and 1TB addons upgrades
- Verify that the buttons behave as expected
- Pick a couple of monthly products in the table below, and check that you see the correct upgrades (one should not be able to upgrade from a monthly to a yearly plan, nor buy the add-on they already own)

| Owned product | Upgrades |
|-----------------|-----------|
| Jetpack VaultPress Backup 10GB (monthly or yearly) | Add-on 10GB, Add-on 100GB, Add-on 1TB |
| Jetpack VaultPress Backup 1TB (monthly or yearly) |	Add-on 1TB, Add-on 3TB, Add-on 5TB |
| Add-on Storage 10GB |	 Add-on 100GB, Add-on 1TB |
| Add-on Storage 100GB | Add-on 1TB, Add-on 3TB |
| Add-on Storage 1TB |Add-on 3TB, Add-on 5TB |
| Add-on Storage 3TB |Add-on 5TB |
| Add-on Storage 5TB | – |
| Security 10GB (monthly or yearly) | Add-on 10GB, Add-on 100GB, Add-on 1TB |
| Security 1TB (monthly or yearly) |	Add-on 1TB, Add-on 3TB, Add-on 5TB |

_Note: there may be some styling issues. This is due to styles not being loaded in the same order locally and in staging. Fixing this is not in the scope of this PR._

### Screenshots

_Before_
<img width="400" alt="Screenshot 2023-03-13 at 3 46 03 PM" src="https://user-images.githubusercontent.com/1620183/224815928-71e1f19e-ea5e-4557-937d-c306812366fc.png">

_After_

<img width="400" alt="Screenshot 2023-03-16 at 8 26 50 AM" src="https://user-images.githubusercontent.com/1620183/225616924-2589817f-08b9-4f9b-8b84-38f195dda91f.png"><img width="400" alt="Screenshot 2023-03-16 at 8 26 57 AM" src="https://user-images.githubusercontent.com/1620183/225616927-13c4053e-b81c-4b2d-bee9-120cd409646f.png">
